### PR TITLE
Update development notes

### PR DESCRIPTION
**Pull request by dmey**
This should make clearer how to release and what to when doing so.

### CODE REVIEW

File: DEVELOP.md [hunk 1]
```diff
@@ -53,38 +105,29 @@ Install the latest version of QGIS 3 using [these](https://gis4wrf.github.io/ins
...
+Before creating a new release, make sure to complete the following checklist:
 
 ```
-# GIS4WRF (https://doi.org/10.5281/zenodo.1288569)
-# Copyright (c) 2018 D. Meyer and M. Riechert. Licensed under MIT.
-```
+- [ ] Create a new issue named `Prepare for GIS4WRF <VERSION_NUMBER> release`, and copy and paste this check-list and tick what is appropriate.
+- [ ] Check that the [documentation](https://gis4wrf.github.io/) repository is up to date.
+- [ ] Check that the [readme](README.md) file is up-to-date.
+- [ ] Check that the [metadata.txt](gis4wrf/metadata.txt) file is up-to-date.
```
Comment (letmaik) [inline DEVELOP.md]: This is duplicate with "Bump the version" below. I would omit this entry and only leave bump the version.

File: DEVELOP.md [hunk 2]
```diff
@@ -53,38 +105,29 @@ Install the latest version of QGIS 3 using [these](https://gis4wrf.github.io/ins
...
-The following header shall be appended at the top of every source file.
+Before creating a new release, make sure to complete the following checklist:
 
 ```
-# GIS4WRF (https://doi.org/10.5281/zenodo.1288569)
-# Copyright (c) 2018 D. Meyer and M. Riechert. Licensed under MIT.
-```
+- [ ] Create a new issue named `Prepare for GIS4WRF <VERSION_NUMBER> release`, and copy and paste this check-list and tick what is appropriate.
+- [ ] Check that the [documentation](https://gis4wrf.github.io/) repository is up to date.
+- [ ] Check that the [readme](README.md) file is up-to-date.
```
Comment (letmaik) [inline DEVELOP.md]: Relative links will not resolve when copied into an issue.

File: DEVELOP.md [hunk 3]
```diff
@@ -53,38 +105,29 @@ Install the latest version of QGIS 3 using [these](https://gis4wrf.github.io/ins
...
+- [ ] Check that the [metadata.txt](gis4wrf/metadata.txt) file is up-to-date.
+- [ ] Create a milestone for the version to release.
```
Comment (letmaik) [inline DEVELOP.md]: This should be done way before releasing. At this point it's more like "if not done already"

File: DEVELOP.md [hunk 4]
```diff
@@ -53,38 +105,29 @@ Install the latest version of QGIS 3 using [these](https://gis4wrf.github.io/ins
...
+- [ ] Add issues and PRs to the milestone.
```
Comment (letmaik) [inline DEVELOP.md]: Same here, this is not really part of releasing but during development.

File: DEVELOP.md (hunk 1 above)
Comment (dmey) [reply DEVELOP.md]: Entry removed

File: DEVELOP.md (hunk 2 above)
Comment (dmey) [reply DEVELOP.md]: Changed to absolute

File: DEVELOP.md (hunk 3 above)
Comment (dmey) [reply DEVELOP.md]: Changed to  "Create a milestone for the version to release if not done already" and moved up

File: DEVELOP.md (hunk 4 above)
Comment (dmey) [reply DEVELOP.md]: Changed to "Check that all issues and PRs targeting the new release have been addded to the milestone." and moved up

**Pull request merged**